File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int,int> mpp;
        for(int i= 0 ;i < nums.size(); i++) {
            mpp[nums[i]]++;
        }
        for(auto it : mpp) {
            if(it.second > nums.size() / 2) {
                return it.first;
            }
        }
        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        Map<Integer,Integer> mpp = new HashMap<>();
        for(int i = 0; i < n; i++) {
            mpp.put(nums[i],mpp.getOrDefault(nums[i],0) +1);
        }
        for(Map.Entry<Integer,Integer> it : mpp.entrySet()) {
            if(it.getValue() > n/2) {
                return it.getKey();
            }
        }
        return -1;
    }
}